### PR TITLE
Fixed Session Timeout

### PR DESCRIPTION
**Pull request by bryanansong**
## Description
In this PR, I fixed the navigation to the login screen when your session times out and also renamed workout statistics to user statistics to provide better context as to what that backend route is for.
<what this pull request is doing>
<what will be done in later PRs and not included here>

## Test Plan

https://github.com/bryanansong/ZenFitness/assets/50881304/fb025574-d797-4146-b7e3-93b8d1b8f95d

<insert images or gifs of feature>
<otherwise, say how code was tested if not UI facing>
<any edge cases you might have specifically tested>

### CODE REVIEW

File: frontend/src/App.tsx [hunk 1]
```diff
@@ -1,20 +1,54 @@
...
+          setIsTokenValid(false);
+        }
+      } else {
+        setIsTokenValid(false);
+      }
+    };
+
+    checkTokenValidity();
+
+    const interval = setInterval(checkTokenValidity, 10000);
```
Comment (m5matthew) [inline frontend/src/App.tsx:40]: Instead of setting an interval, is it possible to just set up a check in the future x seconds after the token expires? Since it seems like we now the token expiration time. Would save client compute.

Note: don't want to you spend too much time on this, we can address this in the future as a stretch

**Review by m5matthew — COMMENTED** [review]
Thanks for handling extra session edge cases!
